src/nizk/bullet.rs

```diff
@@ -168,8 +168,8 @@ impl BulletReductionProof {
 
```

Comment:
(If you wanted to save a few field ops, you could batch invert u^2 rather than squaring both lists. Maybe not worth caring about though?)

Reply:
Yes, I think one could do that, but not sure it affects end-to-end running times much.